src/tasks/select-component.ts

```diff
@@ -0,0 +1,85 @@
+import inquirer from "inquirer";
+import inquirerSearchCheckbox from "inquirer-search-checkbox";
+import { ResourceContext, ZeplinComponent } from "./context/resource";
+import { TaskStep, Task, transitionTo, pauseSpinningAndExecuteTask } from "../util/task";
+import { ComponentSection, Component } from "../api/interfaces";
+import { ZeplinApi } from "../api";
+import * as ui from "./ui/select-component";
+
+inquirer.registerPrompt("search-checkbox", inquirerSearchCheckbox);
+
+const zeplinApi = new ZeplinApi();
+
+function mapComponents(components: Component[]): ZeplinComponent[] {
+    return (components || []).map(c => ({
```

Comment:
We can simplify this line since the type of `components` is an array. Similarly, we simplify the `extractComponents` function

```suggestion
    return components.map(c => ({
```

Reply:
You are correct if components were to be generated by our TS code but it is possible for API response to receive `components` as undefined on runtime theoretically. I think there is no need to interrupt initialization if such a case happens because of a bug on Zeplin API for components of an unrelated component section, so it just safely ignores.

Reply:
If it's possible to get `components` as `undefined` from API, then we should change the type definition of `ComponentSection`. If getting `undefined` is unexpected behaviour, then we should ensure that API does not send `undefined`. This looks like a workaround to me. WDYT?

Reply:
You are right, working on it